apps/pie-microsite/src/_11ty/shortcodes/tokensTable.js

```diff
@@ -0,0 +1,173 @@
+ * @returns {object} an object containing a hexcode and opacity value (if opacity was provided)
+ */
+const splitColorToken = token => {
+    const [hexcode, opacity] = token.split('|');
+
+    return {
+        hexcode,
+        opacity
+    };
+};
+
+/**
+ * Creates the example color swatch to show on the token list item
+ * @param {string} token - the token value i.e. #000, #ffffff, #000|0.85 or #000000|0.85
+ * @returns {string} the color swatch example HTML string
+ */
+const createColorExample = token => {
+    const tokenValues = splitColorToken(token);
+    let cssVariable = `--example-background-color: ${tokenValues.hexcode}`;
+    const classes = ['c-tokensTable-example'];
+
+    if (tokenValues.opacity) {
+        cssVariable = `--example-checked-opacity: ${tokenValues.opacity}`;
+        classes.push('c-tokensTable-example--checked');
+    }
+
+    if (!isColorDark(tokenValues.hexcode)) {
+        classes.push('c-tokensTable-example--bordered');
+    }
+
+    return `<div class="${classes.join(' ')}" style="${cssVariable}";></div>`;
+};
+
+/**
+ * Creates an example element to display in the token list item.
+ * This could be a color swatch, a representation of border radius or spacing etc.
+ * @param {string} token - the token value
+ * @param {string} tokenType - the type of token i.e. color, spacing, radius
+ * @returns {string} the example HTML string
+ */
+const createTokenExampleElement = (token, tokenType) => {
+    switch (tokenType) {
+        case tokenTypes.COLOR:
+            return createColorExample(token);
+        default:
+            throw new Error(`token type not recognised: ${tokenType}. Token:${token}`);
+    }
+};
+
+/**
+ * Creates a token pill element to display the SCSS token name in the token list item.
+ * @param {string} tokenScssName - the text to display
+ * @returns {string} the token pill HTML string
+ */
+const createTokenPill = tokenScssName => `<span class="c-tokensTable-token">${tokenScssName}</span>`;
+
+/**
+ * Creates a token list item element to add to the tokens list.
+ * @param {object} config
+ * @param {string} config.token - the design token value
+ * @param {string} config.tokenType - the type of design token i.e. color, spacing, radius
+ * @param {string} config.tokenScssName - the design token SCSS name i.e. '$color-black'
+ * @param {string} config.tokenDisplayName - the display name of the token i.e. 'Black'
+ * @returns {string} the list item HTML string
+ */
+const createTokenListItem = ({
+    token,
+    tokenType,
+    tokenScssName,
+    tokenDisplayName
+}) => {
+    const tokenPill = createTokenPill(tokenScssName);
+    const tokenExampleElement = createTokenExampleElement(token, tokenType);
+
+    return `<li class="c-tokensTable-row c-tokensTable-item">
+      ${tokenExampleElement}
+      <div class="c-tokensTable-content">
+        <span class="c-tokensTable-displayName">${tokenDisplayName}</span>
+        <span></span>
+        <span></span>
```

Comment:
What are those empty spans for?

Reply:
We had some copy in there initially but chose to take it out and move it into another PR.
These were left in as placeholders, but aren't needed. I've removed them now
Thanks 🙂